src/swagger/openapi.json

```diff
@@ -0,0 +1,287 @@
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/api/authenticate": {
+      "post": {
+        "tags": [
+          "authentication"
+        ],
+        "summary": "Authenticate a user",
+        "operationId": "authenticateUser",
+        "requestBody": {
+          "description": "login and password passed in req.body",
+          "content": {
+            "application/json": {
+              "schema": {
+                "type": "object",
+                "properties": {
+                  "login": {
+                    "type": "string"
+                  },
+                  "password": {
+                    "type": "string"
+                  }
+                }
+              }
+            }
+          },
+          "required": true
+        },
+        "responses": {
+          "200": {
+            "description": "User authenticated",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object",
+                  "properties": {
+                    "token": {
+                      "type": "string"
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "User was found in the database, but the password given in the req.body doesn't match what is in the database.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "No user with that username was found in the database.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/api/logout": {
+      "post": {
+        "tags": [
+          "logout"
+        ],
+        "summary": "Log a user out",
+        "operationId": "logoutUser",
+        "responses": {
+          "200": {
+            "description": "User logged out"
+          },
+          "401": {
+            "description": "Invalid user session",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/api/articles": {
+      "post": {
+        "tags": [
+          "articles"
+        ],
+        "summary": "Create an article",
+        "operationId": "createArticle",
+        "requestBody": {
+          "description": "Created article object",
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/Article"
```

Comment:
When a request comes in to create an article, it requires a token parameter as a header, and content, visibility, title, and article_id in the body. This schema reference includes a `login` which is not used here, so either update the schema or don't refer to the schema and just put the correct parameters here. `$ref` should really only be used if there are repeated objects. We don't have that, since the POST body has different properties than the GET response.

Reply:
I removed the Article schema and just set it to an object with the parameters.